app/controllers/user_reservations_controller.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class UserReservationsController < ApplicationController
+  include JSONAPI::Fetching
+
+  def index
+    reservations = current_user.reservations
+
+    render jsonapi: reservations, except: blacklisted_attributes
+  end
+
+  def show
+    reservation = current_user.reservations.find(params[:id])
```

Comment:
You can handle that through pundit as well - again hawing single controller instead of two. In the Policy, you would return `true` always for employees, for users, you would return true only if `user.id == reservations.user_id` - that way users would have access only to their own reservations and employees to all of them.